### PR TITLE
feat(client): integrate relation joins

### PR DESCRIPTION
**Pull request by aqrln**
Integration PR for https://github.com/prisma/prisma-engines/pull/4509

### CODE REVIEW

File: packages/client/package.json [hunk 1]
```diff
@@ -96,15 +96,15 @@
     "@prisma/debug": "workspace:*",
     "@prisma/driver-adapter-utils": "workspace:*",
     "@prisma/engines": "workspace:*",
-    "@prisma/engines-version": "5.7.0-39.0d724e37b36bf276aa095d4571213cd035eafad0",
+    "@prisma/engines-version": "5.7.0-38.integration-join-support-8109c3fa441c3c6797ed19f327d9d667b102ab9a",
```
Comment (aqrln) [inline packages/client/package.json:99]: Revert before merging

File: .github/workflows/test-template.yml [hunk 2]
```diff
@@ -286,6 +287,11 @@ jobs:
         flavor: ['js_pg', 'js_neon', 'js_libsql', 'js_planetscale']
         shard: ['1/6', '2/6', '3/6', '4/6', '5/6', '6/6']
         node: [20] #[16, 18, 20]
+        previewFeatures: ['driverAdapters', 'driverAdapters,relationJoins']
+        exclude:
+          # Exclude relationJoins tests with WASM because the WASM engine does not include the necessary changes yet.
+          - engine-type: 'wasm'
+            previewFeatures: 'driverAdapters,relationJoins'
```
Comment (millsp) [inline .github/workflows/test-template.yml:294]: Could you please create a small ticket for us to get back to this later (once wasm engine publish/update is streamlined).

File: packages/client/tests/functional/0-legacy-ports/batch-find-unique/tests.ts [hunk 3]
```diff
@@ -76,7 +78,47 @@ testMatrix.setupTestSuite(() => {
       }
     })
 
-    expect(executedBatchQuery).toMatchSnapshot()
```
Comment (millsp) [inline packages/client/tests/functional/0-legacy-ports/batch-find-unique/tests.ts:79]: Alternatively, I think snapshots can be named via the first optional argument. That said, I kinda prefer the inline version.

File: .github/workflows/test-template.yml [hunk 4]
```diff
@@ -165,7 +166,7 @@ jobs:
           skip-tsc: false
 
       # Run the functional test suite
-      - run: pnpm run test:functional --silent --shard ${{ matrix.shard }} --engine-type ${{ matrix.queryEngine }}
+      - run: pnpm run test:functional --silent --shard ${{ matrix.shard }} --engine-type ${{ matrix.queryEngine }} --preview-features "${{ matrix.previewFeatures }}"
```
Comment (millsp) [inline .github/workflows/test-template.yml:169]: Unrelated: It would be nice to have this kind of testing for all preview features at once.

**Review by millsp — APPROVED** [review]

File: .github/workflows/test-template.yml (hunk 4 above)
Comment (millsp) [reply .github/workflows/test-template.yml:169]: (And not required now)

File: .github/workflows/test-template.yml (hunk 2 above)
Comment (millsp) [reply .github/workflows/test-template.yml:294]: https://github.com/prisma/team-orm/issues/697

**Pull request merged**